lavalink/PlayerManager.py

```diff
@@ -206,6 +209,32 @@ def add_at(self, index: int, requester: int, track: dict):
             self.volume = max(min(vol, 1000), 0)
         await self.node.ws.send(op='volume', guildId=self.guild_id, volume=self.volume)
 
+    async def set_gain(self, band: int, gain: float = 0.0):
+        """ (Only Lavalink v3.1 or higher) Sets the equalizer band (0-15) gain to the given amount.
+        A gain of 0.0 indicates no change. Gain cannot go below -0.25, or exceed 1.0 """
+        if not self.node.server_version == 3 and not self.node.ws._is_v31:
+            return
+        gain = max(min(gain, 1.0), -0.25)
+        band = max(min(band, 15), 0)
+        self.equalizer[band] = gain
+        await self.node.ws.send(op='equalizer', guildId=self.guild_id, bands=[{'band': band, 'gain': gain}])
+
+    async def bulk_set_gain(self, gain_list: list):
+        """ (Only Lavalink v3.1 or higher) Sets equalizer to the specified values in the list. Must have 16 values. """
+        if not self.node.server_version == 3 and not self.node.ws._is_v31:
+            return
+        if len(gain_list) != 16:
+            raise ValueError('gain_list must have exactly 16 values')

```

Comment:
Lavalink doesn't worry about how many bands you pass. Perhaps this should accept >0 and <=16 bands? I.e. if I wanted to make 2-3 changes to bands for bass boost I'd rather make 1 call versus 3 calls, or having to provide an entire array of floats

Reply:
So we make a dictionary with key for the band and value for the gain OR list of tuples `[(0, 0.0), (1, -0.20), (5, 0.75)]` ?

Reply:
Hmm how about a Equalizer class like a EQ factory where you can define a class and use methods like add(band:int, gain:float) etc

Reply:
Can confirm that Lavalink doesn't worry about how many bands you pass 

![image](https://user-images.githubusercontent.com/19573589/47606888-bf1a2780-da19-11e8-82c7-0245437f0a6e.png)

Reply:
what happens when you send more than 16 bands?
like `[...{"band": 16, "gain": 0.25}]`

Reply:
Lavaplayer checks if the band is valid and applies the gain for that band if it is, otherwise it's ignored. [Relevant code](https://github.com/sedmelluq/lavaplayer/blob/master/main/src/main/java/com/sedmelluq/discord/lavaplayer/filter/equalizer/EqualizerConfiguration.java#L23)

Reply:
That being said you will probably need to do some filtering within the client to ensure that only valid values are retained.

Reply:
I wouldn't raise an error if band was invalid, but I'd just to a `log.warning('Passed band index is out of range ({}) min: 0, max: 15'.format(band))`

Reply:
kek why warning, it's not that important